ext/standard/pack.c

```diff
@@ -864,13 +874,6 @@ PHP_FUNCTION(unpack)
 				RETURN_THROWS();
 		}
 
-		if (size != 0 && size != -1 && size < 0) {
-			php_error_docref(NULL, E_WARNING, "Type %c: integer overflow", type);
-			zend_array_destroy(Z_ARR_P(return_value));
-			RETURN_FALSE;
-		}
```

Comment:
There's an `size = (repetitions > 0) ? (repetitions + (repetitions % 2)) / 2 : repetitions;` assignment above. If repetitions is INT_MAX then repetitions + (repetitions % 2) would be INT_MIN and size would be INT_MIN/2.